src/main/java/org/example/movieSpring/service/MovieService.java

```diff
@@ -0,0 +1,51 @@
+package org.example.movieSpring.service;
+
+import org.example.movieSpring.model.Movie;
+import org.example.movieSpring.repo.MovieRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieService {
+
+    private final MovieRepo movieRepo;
+
+    @Autowired
+    public MovieService(MovieRepo movieRepo) {
+        this.movieRepo = movieRepo;
+    }
+
+    public List<Movie> getMovies() {
+        return movieRepo.findAll();
+    }
+
+    public Optional<Movie> getMovieById(Long id) {
+        return movieRepo.findById(id);
+    }
+
+    public Movie addMovie(Movie movie) {
+        return movieRepo.save(movie);
+    }
+
+    public List<Movie> getMoviesByReleaseYear(int releaseYear) {
+        return movieRepo.findAll().stream().filter(m -> m.getRelease_year() == releaseYear).toList();
+    }
+
+    public Movie updateMovie(Movie movie) {
+        Optional<Movie> newMovie = getMovieById(movie.getId());
+        if (newMovie.isEmpty()) {
+            throw new NullPointerException("Could not find object with that ID");
+        }
+        newMovie.get().setTitle(movie.getTitle());
+        newMovie.get().setRelease_year(movie.getRelease_year());
+        return movieRepo.save(newMovie.get());
```

Comment:
this is not how optionals are used. 
an ifPresent() . orElse() construct would be better.

why does the program throw an exception because someone else sends an id that did not exist via a rest api? is it an error on our side side? can we recover from this?

Reply:
Changed how it functions